File: 1154-day-of-the-year/1154-day-of-the-year.java
class Solution {
    public int dayOfYear(String date) {
        int[] monthsum = {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334};
        int ans = 0;
        int year = 1000*((int)date.charAt(0)-'0')+100*((int)date.charAt(1)-'0')+10*((int)date.charAt(2)-'0')+((int)date.charAt(3)-'0');
        int month = 10*((int)date.charAt(5)-'0')+((int)date.charAt(6)-'0');
        int day = 10*((int)date.charAt(8)-'0')+((int)date.charAt(9)-'0');
        if(year%4==0 && year != 1900 && month>2){
            monthsum[month-1]++;
        }
        ans=day+monthsum[month-1];
        return ans;
    }
}

File: 1154-day-of-the-year/1154-day-of-the-year.py
class Solution:
    def dayOfYear(self, date: str) -> int:
        year, month, day = map(int, date.split('-'))
        days_month_wise = [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334]
        
        if year%4 == 0 and year/100 != 19:
            for i in range(2, len(days_month_wise)):
                days_month_wise[i] += 1
        
        ans = days_month_wise[month-1] + day
        
        return ans